tests/simplex/grid_out_msh.cc

```diff
@@ -0,0 +1,66 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Create a triangulation with simplices and output it in gmsh format
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/tria.h>
+
+#include <fstream>
+
+#include "../tests.h"
+
+template <int dim>
+void
+check()
+{
+  Triangulation<dim> triangulation;
+
+  {
+    Triangulation<dim> x;
+    GridGenerator::subdivided_cylinder(x, 4, 1, 2);
+
+    // For the cylinder geometry
+    for (auto cell : x.active_cell_iterators())
+      for (const unsigned int face : cell->face_indices())
+        if (cell->at_boundary(face))
+          if (std::fabs(cell->face(face)->center()[0] - 0) <
+              0.1 * cell->diameter())
+            cell->face(face)->set_boundary_id(1);
+          else if (std::fabs(cell->face(face)->center()[0] - 4) <
+                   0.1 * cell->diameter())
+            cell->face(face)->set_boundary_id(2);
```

Comment:
I get an ambiguous brace warning here:
```suggestion
          {
            if (std::fabs(cell->face(face)->center()[0] - 0) <
                0.1 * cell->diameter())
              cell->face(face)->set_boundary_id(1);
            else if (std::fabs(cell->face(face)->center()[0] - 4) <
                     0.1 * cell->diameter())
              cell->face(face)->set_boundary_id(2);
           }
```